src/components/FileUpload/FileUpload.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import Dropzone from 'react-dropzone';
+import PropTypes from 'prop-types';
+
+import uploadArrow from './upload_arrow.png';
+import complete from './complete.png';
+
+import styled from 'styled-components';
+
+import ProgressBar from './ProgressBar';
+import Card from './Card';
+import LinkButton from './LinkButton';
+import classnames from 'classnames';
+
+const UploadImg = styled.img`
+  width: 85px;
+  display: block;
+  margin: 10px auto;
+`;
+
+const P = styled.p`
+  margin: 0;
+  text-align: center;
+`;
+
+const ActionMessage = styled(P)`
+  font-weight: bold;
+`;
+
+const ErrorMessage = styled(P)`
+  font-size: 13px;
+  color: #ff3b3b;
+`;
+const ManualMessage = styled(P)`
+  color: #aaaaaa;
+  display: inline;
+`;
+
+const DropZoneFile = styled(Dropzone)`
+  display: inline;
+  color: #3b86ff;
+  text-decoration: underline;
+  cursor: pointer;
+`;
+
+const CenterDiv = styled.div`
+  text-align: center;
+`;
+
+const StyledCard = styled(Card)`
+  height: 150px;
+  &.disabled {
+    opacity: 0.5;
+  }
+`;
+
+FileUpload.EMPTY = 'EMPTY';
+FileUpload.UPLOADING = 'UPLOADING';
+FileUpload.COMPLETED = 'COMPLETED';
+FileUpload.UPLOAD_ERROR = 'UPLOAD_ERROR';
+FileUpload.FILE_TYPE_ERROR = 'FILE_TYPE_ERROR';
+
+const propTypes = {
+  disabled: PropTypes.bool,
+  onDropAccepted: PropTypes.func.isRequired,
+  onDropRejected: PropTypes.func.isRequired,
+  onRetry: PropTypes.func.isRequired,
+  progress: PropTypes.number.isRequired,
+  status: PropTypes.oneOf([
+    FileUpload.COMPLETED,
+    FileUpload.EMPTY,
+    FileUpload.UPLOADING,
+    FileUpload.UPLOAD_ERROR,
+    FileUpload.FILE_TYPE_ERROR
+  ])
+};
+
+const defaultProps = {
+  progress: 0,
+  status: FileUpload.EMPTY
+};
+
+function FileUpload({ disabled, status, onDropAccepted, onDropRejected, progress, onRetry }) {
+  const handleDropAccepted = (event) => {
+    onDropAccepted(event[0]);
+  };
+
+  const handleDropRejected = (event) => {
+    onDropRejected(event);
+  };
+
+  const UploadImage = <UploadImg alt="upload arrow" src={uploadArrow} />;
+  const cardClassNames = classnames({ disabled: disabled });
```

Comment:
```suggestion
  const cardState = disabled ? 'disabled' : 'ok';
```
and then you just pass state down